File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);
const board = input.slice(1);
const count = [];

for (let i = 0; i < N - 7; i++) {
    for (let j = 0; j < M - 7; j++) {
        let wCount = 0;
        let bCount = 0;
        
        for (let di = i; di < i + 8; di++) {
            for (let dj = j; dj < j + 8; dj++) {
                if ((di + dj) % 2 === 0) {
                    if (board[di][dj] !== 'W') {
                        wCount++;
                    } else {
                        bCount++;
                    }
                } else {
                    if (board[di][dj] !== 'W') {
                        bCount++;
                    } else {
                        wCount++;
                    }
                }
            }
        }
        
        count.push(wCount);
        count.push(bCount);
    }
}

console.log(Math.min(...count));

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys 

N, M = map(int, input().split())

board = sys.stdin.read().split()
count = []

for i in range(N - 7):
    for j in range(M - 7):
        w_count = 0
        b_count = 0
        
        for di in range(i, i + 8):
            for dj in range(j, j + 8):
                if (di + dj) % 2 == 0:
                    if board[di][dj] != 'W':
                        w_count += 1
                    else:
                        b_count += 1
                else:
                    if board[di][dj] != 'W':
                        b_count += 1
                    else:
                        w_count += 1
        
        count.append(w_count)
        count.append(b_count)

print(min(count))